python/lsst/ts/observatory/control/mock/mtcs_async_mock.py

```diff
@@ -461,6 +470,33 @@ async def mtdome_tel_light_wind_screen_next(
     ) -> types.SimpleNamespace:
         return self._mtdome_tel_light_wind_screen
 
+    async def mtdome_cmd_park(self, *args: typing.Any, **kwargs: typing.Any) -> None:
+        asyncio.create_task(self._mtdome_park())
+
+    async def _mtdome_park(self, *args: typing.Any, **kwargs: typing.Any) -> None:
+        # Mock implementation of cmd_park
+        await asyncio.sleep(self.heartbeat_time)
+        self.log.info("Dome park command executed")
+        self._mtdome_evt_azMotion_state = types.SimpleNamespace(
+            state=MTDome.MotionState.PARKED, inPosition=True
+        )
+
+    async def mtdome_evt_az_motion_state_next(
+        self, *args: typing.Any, **kwargs: typing.Any
+    ) -> types.SimpleNamespace:
+        return await asyncio.create_task(
```

Comment:
you don't need to create a task here, just do:

```
await asyncio.sleep(self.heartbeat_time * 3)
return self._mtdome_evt_azMotion_state
```

and remove the `_mtdome_evt_az_motion_state_next` method.

Reply:
Creating a task was the only way to avoid a race condition, if I don't do a task, the event seems to set to the original state. We need the state to change from one command, say park to another say unpark and the only way I managed that was via a task.

Reply:
34b8c780a51499a3402d50cd51c01ff6b71c7ded